subsys/bluetooth/host/hci_core.c

```diff
@@ -5432,16 +5445,71 @@ static void le_read_buffer_size_complete(struct net_buf *buf)
 
 	BT_DBG("status 0x%02x", rp->status);
 
-	bt_dev.le.mtu = sys_le16_to_cpu(rp->le_max_len);
-	if (!bt_dev.le.mtu) {
+	bt_dev.le.acl_mtu = sys_le16_to_cpu(rp->le_max_len);
+	if (!bt_dev.le.acl_mtu) {
 		return;
 	}
 
-	BT_DBG("ACL LE buffers: pkts %u mtu %u", rp->le_max_num, bt_dev.le.mtu);
+	BT_DBG("ACL LE buffers: pkts %u mtu %u", rp->le_max_num,
+	       bt_dev.le.acl_mtu);
 
-	k_sem_init(&bt_dev.le.pkts, rp->le_max_num, rp->le_max_num);
+	k_sem_init(&bt_dev.le.acl_pkts, rp->le_max_num, rp->le_max_num);
 }
-#endif
+
+static void read_buffer_size_v2_complete(struct net_buf *buf)
+{
+#if defined(CONFIG_BT_ISO)
+	struct bt_hci_rp_le_read_buffer_size_v2 *rp = (void *)buf->data;
+	uint8_t max_num;
+
+	BT_DBG("status %u", rp->status);
+
+	bt_dev.le.acl_mtu = sys_le16_to_cpu(rp->acl_mtu);
+	if (!bt_dev.le.acl_mtu) {
+		return;
+	}
+
+	BT_DBG("ACL LE buffers: pkts %u mtu %u", rp->acl_max_pkt,
+		bt_dev.le.acl_mtu);
+
+	max_num = MIN(rp->acl_max_pkt, CONFIG_BT_CONN_TX_MAX);
+	k_sem_init(&bt_dev.le.acl_pkts, max_num, max_num);
+
+	bt_dev.le.iso_mtu = sys_le16_to_cpu(rp->iso_mtu);
+	if (!bt_dev.le.iso_mtu) {
+		return;
```

Comment:
Is this an error? Can you return it, or signal it somehow?

Reply:
Or perhaps at least have `BT_DBG` or higher level log statement